apollo-openapi/src/main/java/com/ctrip/framework/apollo/openapi/client/ApolloOpenApiClient.java

```diff
@@ -115,36 +121,54 @@ private ApolloOpenApiClient(String portalUrl, String token, RequestConfig reques
    *
    * @since 1.5.0
    */
+  @Override
   public OpenClusterDTO getCluster(String appId, String env, String clusterName) {
     return clusterService.getCluster(appId, env, clusterName);
   }
 
   /**
-   * Create the cluster
-   *
+   * Create the cluster. Suggest use {@link #createCluster(String, String, OpenClusterDTO)} instead.
    * @since 1.5.0
    */
   public OpenClusterDTO createCluster(String env, OpenClusterDTO openClusterDTO) {
+    return this.createCluster(openClusterDTO.getAppId(), env, openClusterDTO);
+  }
+
+  @Override
+  public OpenClusterDTO createCluster(String appId, String env, OpenClusterDTO openClusterDTO) {
+    if (null == appId) {
+      throw new IllegalArgumentException("appId cannot be null");
+    }
+    if (!appId.equals(openClusterDTO.getAppId())) {
+      throw new IllegalArgumentException("appId should same as in OpenClusterDTO");
+    }
     return clusterService.createCluster(env, openClusterDTO);
   }
 
   /**
    * Get the namespace
    */
+  @Override
   public OpenNamespaceDTO getNamespace(String appId, String env, String clusterName, String namespaceName) {
     return namespaceService.getNamespace(appId, env, clusterName, namespaceName);
   }
 
   /**
-   * Create the app namespace
+   * Create the app namespace. Suggest use {@link #createNamespace(String, OpenAppNamespaceDTO)} instead.
    */
   public OpenAppNamespaceDTO createAppNamespace(OpenAppNamespaceDTO appNamespaceDTO) {
     return namespaceService.createAppNamespace(appNamespaceDTO);
   }
 
+  @Override
+  public OpenAppNamespaceDTO createNamespace(String appId, OpenAppNamespaceDTO appNamespaceDTO) {
```

Comment:
How about keeping the original method name?
```suggestion
  public OpenAppNamespaceDTO createAppNamespace(String appId, OpenAppNamespaceDTO appNamespaceDTO) {
```